torch/nn/attention/flex_attention.py

```diff
@@ -845,7 +845,7 @@ def causal_mask(b, h, q_idx, kv_idx):
         Q_LEN = _round_up_to_multiple(Q_LEN, Q_BLOCK_SIZE)
     KV_LEN = _round_up_to_multiple(KV_LEN, KV_BLOCK_SIZE)
     if _compile:
-        inner_func = torch.compile(inner_func, fullgraph=True, dynamic=False)
+        inner_func = torch.compile(inner_func, fullgraph=True, dynamic=True)
```

Comment:
Don't add `dynamic=True` - just remove the dynamic setting. Allow for automatic dynamic shapes I think.